historical_surveillance/views.py

```diff
@@ -0,0 +1,328 @@
+                stu = {
+                    "error_message": error_message,
+                    "districts_names": districts_names,
+                    "district_selected" : district_selected,
+                    "data" : data,
+                }
+                return render(request, 'district_1.html', stu)
+        else:
+            return render(request, 'district_1.html', {"districts_names":districts_names})
+
+
+
+    else:
+        error_message = "No records found for the selected district"
+
+
+
+def district_1_trend(request):
+    error_message = None
+    graph = None
+
+    district_df = pd.DataFrame(District.objects.all().values())
+    districts_lists = district_df['District_Name']
+
+    schools_df = pd.DataFrame(AggregateEnrollment.objects.all().values())
+    schools = schools_df['name_of_school'].unique()
+
+    data = AggregateEnrollment.objects.all().values('name_of_school', 'category_of_school', 'district_of_school').distinct().\
+        annotate(total=Count('name_of_school')).order_by('category_of_school')
+
+    if schools_df.shape[0] > 0:
+        if request.method == 'POST':
+            chart = request.POST.get('chart', False)
+            selected_school = request.POST.get('school_name', None)
+            if not chart:
+                if not selected_school:
+                    error_message = "Please select a chart type and school"
+
+            else:
+                if chart is not False:
+                    if selected_school is not None:
+                        select_school_df = pd.DataFrame(AggregateEnrollment.objects.all().
+                                                        filter(name_of_school=selected_school).values())
+                        select_school_df.sort_values(by=['academic_year'], inplace=True)
+                        select_school_df = select_school_df.tail(10)
+                        # function to get the graph
+                        graph = get_plot(chart, z=select_school_df['capacity_of_school'],
+                                         y=select_school_df['total_enrollment'],
+                                         x=select_school_df['academic_year'],
+                                         data=select_school_df,
+                                         name_of_school=selected_school)
+    else:
+        error_message = "No records found"
+
+    stu = {
+        "graph": graph,
+        "error_message": error_message,
+        "name_of_school": data,
+        "schools": schools,
+        "schools_df": schools_df,
+        "districts_lists" : districts_lists,
+
+    }
+    return render(request, 'district_1_trend.html', stu)
+
+def compare_trends(request):
+    error_message = None
+    graph = None
+    district_df = pd.DataFrame(District.objects.all().values())
+    year_df = pd.DataFrame(AggregateEnrollment.objects.all().values())
+    districts_lists = district_df['District_Name']
```

Comment:
`districts_lists = District.objects.values_list('district_name', flat=True).distinct()`

So we don't need to load all districts data in memory and into a dataframe